### PR TITLE
Fix link args for cusolver

### PR DESCRIPTION
**Pull request by unnonouno**
link_args is required to use cusolver.

### CODE REVIEW

**Comment by takagi** [comment]
I want to know more about this change. `-fopenmp` isn't compile time option? for link time?

File: cupy_setup_build.py [hunk 1]
```diff
@@ -270,13 +270,16 @@ def make_extensions(options, compiler, use_cython):
...
             # openmp is required for cusolver
             if compiler.compiler_type == 'unix' and sys.platform != 'darwin':
                 # In mac environment, openmp is not required.
-                args.append('-fopenmp')
+                compile_args.append('-fopenmp')
+                link_args.append('-fopenmp')
             elif compiler.compiler_type == 'msvc':
-                args.append('/openmp')
+                compile_args.append('/openmp')
+                link_args.append('/openmp')
```
Comment (takagi) [inline cupy_setup_build.py]: Would you remove this line for windows so that msvc does not make a warning?

Comment (unnonouno) [inline cupy_setup_build.py]: ok.

**Comment by takagi** [comment]
jenkins, test this please.

**Comment by takagi** [comment]
As @unnonouno says we add compile time and link time options on ubuntu, and as @kmatsuura says we add compile time option only on Windows. At least, it should not break the build. I will merge this PR after I will check CI passing.

**Comment by takagi** [comment]
LGTM!

**Pull request merged**